include/ardpolyfills/BoardData.hxx

```diff
@@ -0,0 +1,52 @@
+/*
+ *  BoardData.hxx
+ *  Copyright 2020 ItJustWorksTM
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+
+#ifndef SMARTCAR_EMUL_BOARDDATA_HXX
+#define SMARTCAR_EMUL_BOARDDATA_HXX
+#include <atomic>
+#include <range\v3\range\conversion.hpp>
+
+template <std::size_t N>
+
+struct FixBufferBus {
+    std::array<std::byte, N> rx;
+    int len1;
+    std::array<std::byte, N> tx;
+    int len2;
+};
+
+struct DynaBufferBus {
+    std::vector<std::byte> rx;
+    std::vector<std::byte> tx;
+};
+    
+struct UartBus : DynaBufferBus {};
+
+struct I2cBus : FixBufferBus<32u> {};
+
+struct BoardData {
+    std::atomic_bool digital_pin_values[32];
+    std::atomic_uint16_t analog_pin_values[32];
+    std::atomic_uint8_t servo_value[10]; // �\_(?)_/� 
+    std::atomic_uint8_t pin_frequency[32];
```

Comment:
Should be vectors, since different boards have different pin counts
As a general guideline, whenever you have arrays, use at least `std::array` instead of C-style arrays